shopfloor_workstation/services/workstation.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2021 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+
+
+class ShopfloorWorkstation(Component):
+    """
+    Workstation service for the mobile application.
+
+    This allows to scan a workstation bar code on the floor, which
+    will set a default profile on the mobile app and offer the option
+    to change some configuration of the connected user.
+
+    """
+
+    _inherit = "base.shopfloor.service"
+    _name = "shopfloor.workstation"
+    _usage = "workstation"
+    _description = __doc__
+
+    def setdefault(self, barcode):
+        """Endpoint that receives a scanned barcode."""
+        ws = self.env["shopfloor.workstation"].search([("barcode", "=", barcode)])
+        if ws:
+            ws.set_as_default_on_user(self.env.user)
+            message = {
+                "message_type": "info",
+                "body": _("Default workstation set to {}").format(ws.name),
+            }
+        else:
+            message = {
+                "message_type": "error",
+                "body": _("Workstation not found"),
+            }
+        return self._response(
+            message=message, data={"size": len(ws), "records": self._to_json(ws)}
```

Comment:
why do you return `records`? Here you should have a single record (the workstation) or nothing.
And you don't need `self._to_json`. Just run `_convert_one_record` or `ws.jsonify($parser, one=True)`